mlfinlab/feature_importance/fingerpint.py

```diff
@@ -0,0 +1,259 @@
+
+        self.feature_column_position_mapping = dict(zip(self.X.columns, range(0, self.X.shape[1])))
+
+        # Get partial dependency functions
+        self._get_individual_partial_dependence()
+
+    @abstractmethod
+    def _get_model_predictions(self, X_):
+        """
+        Get model predictions based on problem type (predict for regression, predict_proba for classification)
+
+        :param X_: (np.array) feature set
+        :return: (np.array) of predictions
+        """
+        raise NotImplementedError('Must implement _get_model_predictions')
+
+    def _get_individual_partial_dependence(self) -> None:
+        """
+        Get individual partial dependence function values for each column.
+
+        :return: None
+        """
+        for col in self.X.columns:
+            y_mean_arr = []  # Array of mean(prediction) for each feature value
+            for x_k in self.feature_values[col]:
+                # Step 2
+                col_k_position = self.feature_column_position_mapping[col]
+                X_ = self.X.values.copy()
+                X_[:, col_k_position] = x_k
+
+                # Step 3
+                y_pred = self._get_model_predictions(X_)
+                y_pred_mean = np.mean(y_pred)
+
+                y_mean_arr.append(y_pred_mean)
+
+            self.ind_partial_dep_functions[col] = y_mean_arr
+
+    def _get_linear_effect_estimation(self) -> dict:
+        """
+        Get linear effect estimates as the mean absolute deviation of the linear predictions around their average value.
+
+        :return: (dict) of linear effect estimates for each feature column.
+        """
+        store = {}
+        for col in self.X.columns:
+            x = self.feature_values[col].reshape(-1, 1)
+            y = self.ind_partial_dep_functions[col]
+
+            lmodel = LinearRegression(fit_intercept=True, normalize=False)
+            lmodel.fit(x, y)
+            y_mean = np.mean(y)
+            linear_effect = np.mean(np.abs(lmodel.predict(x) - y_mean))
+            store[col] = linear_effect
+        return store
+
+    def _get_non_linear_effect_estimation(self) -> dict:
+        """
+        Get non-linear effect estimates as as the mean absolute deviation of the total marginal (single variable) effect
+        around its corresponding linear effect.
+        :return: (dict) of non-linear effect estimates for each feature column
+        """
+        store = {}
+        for col in self.X.columns:
+            x = self.feature_values[col].reshape(-1, 1)
+            y = self.ind_partial_dep_functions[col]
+
+            lmodel = LinearRegression(fit_intercept=True, normalize=False)
+            lmodel.fit(x, y)
+            nonlinear_effect = np.mean(np.abs(lmodel.predict(x) - y.values))
+            store[col] = nonlinear_effect
+        return store
+
+    def get_pairwise_effect(self, combinations: list) -> dict:
```

Comment:
I still like that we have this function so that the end-user can compute the pairwise effects on an Adhoc basis as it requires more computation power.

However, this method doesn't return the pairwise effects but rather stores them in the attribute `self.pairwise_effect`. It is for that reason that I think the name should be changed to `compute_pairwise_effect`. If we keep it as a get method then it should also return the pairwise effects.

Reply:
Good idea